pandas/io/formats/csvs.py

```diff
@@ -188,6 +198,42 @@ def save(self):
                 for _fh in handles:
                     _fh.close()
 
+    def _index_label_encoder(self):
+        """
+        Encode index label if it is not False.
+
+        Returns
+        -------
+        index_label: list
+            New index_label given index types
+        encode_labels: list
+            List of index labels
+        """
+        index_label = self.index_label
+        obj = self.obj
+
+        if index_label is True:
+            index_label = []
+            # append index label based on index type
+            if isinstance(obj.index, ABCMultiIndex):
+                for name in obj.index.names:
+                    # add empty string is name is None
+                    if name is None:
+                        name = ''
+                    index_label.append(name)
+            else:
+                # if no name, use empty string
+                if obj.index.name is None:
```

Comment:
i don't htink you need this branch at all

Reply:
i think it's needed, and looks like if the branch is removed, lots of tests will fail

Reply:
see above the cases for multiindex and index are the same.